src/include/migraphx/algorithm.hpp

```diff
@@ -95,20 +95,24 @@ inline size_t levenshtein_distance(const std::string& s1, const std::string& s2)
     const size_t l1 = s1.length();
     const size_t l2 = s2.length();
 
-    std::vector<std::vector<size_t>> d(l1 + 1, std::vector<size_t>(l2 + 1));
+    std::vector<size_t> d0(l2 + 1), d1(l2 + 1);
 
-    for(size_t i = 1; i <= l1; i++)
-        d[i][0] = i;
     for(size_t j = 1; j <= l2; j++)
-        d[0][j] = j;
+        d0[j] = j;
 
     for(size_t i = 1; i <= l1; i++)
+    {
+        d1[0] = i;
```

Comment:
I dont think you need a seperate vector for this. You could just store the previous distance in a variable and read that:

```cpp
for (size_t i = 1; i <= l1; i++)
{
    size_t previous = 0;
    for (size_t j = 1; j <= l2; j++)
    {
        size_t cost_insert_or_delete = std::min(d0[j - 1] + 1, d0[j] + 1);
        size_t cost_substitute       = previous + (s1[i - 1] == s2[j - 1] ? 0 : 1);
        previous                     = d0[j];
        d0[j]                        = std::min(cost_substitute, cost_insert_or_delete);
    }
}
```

It seems simpler than swapping the vector at each loop.

Reply:
> It seems simpler than swapping the vector at each loop.

@pfultz2, I am not so sure if a vector swap (which is effectively just an exchange of its internal buffer pointers, and is performed outside the loop) is any less simpler than some operation inside the inner loop?  That algorithm original looks more understandable, and is intuitively what the flow-chart would be. 

I grant you it takes two vectors instead of one. 

But, if you strongly dislike it, I can change it, else leave it as it is for algorithmic simplicity/clarity.. I don't think performance is an issue here -- well it is in the original algorithm.. N**3 vs N**2.

Reply:
> is performed outside the loop)

Its not performed in the inner loop but it is still performed in the outer loop. 

> That algorithm original looks more understandable, and is intuitively what the flow-chart would be.

It took quite bit of work to realize that you are just using the previous distance. Its not very obvious. Maybe that matches the pseudo algorithm on wikipedia, but using the `previous` variable makes what the algorithm is actually doing in the code much clearer. Even rosetta code uses only one array(although the implementation is different):

https://rosettacode.org/wiki/Levenshtein_distance#Generic_ISO_C++_version

> But, if you strongly dislike it, I can change it,

I would prefer to use a single vector unless my analysis of the loop is incorrect(which is quite possible as well).